### PR TITLE
HBASE-23253 Rewrite rsgroup related UTs with the new methods introduc…

### PR DESCRIPTION
**Pull request by Apache9**
…ed in HBASE-22932

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 18 new or modified test files.  |
||| _ HBASE-22514 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 48s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 15s |  HBASE-22514 passed  |
| +1 :green_heart: |  compile  |   2m 57s |  HBASE-22514 passed  |
| +1 :green_heart: |  checkstyle  |   2m 31s |  HBASE-22514 passed  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 53s |  HBASE-22514 passed  |
| +0 :ok: |  spotbugs  |   1m 17s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 15s |  branch/hbase-annotations no findbugs output file (findbugsXml.xml)  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  1s |  the patch passed  |
| +1 :green_heart: |  cc  |   3m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  1s |  the patch passed  |
| -1 :x: |  checkstyle  |   2m 33s |  root: The patch generated 2 new + 60 unchanged - 1 fixed = 62 total (was 61)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   4m 43s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 40s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   8m 20s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 54s |  the patch passed  |
| +0 :ok: |  findbugs  |   0m 13s |  hbase-annotations has no data from findbugs  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  10m  9s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 133m 31s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestInterfaceAlign |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/813 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux bab097f5cd9b 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-813/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / eb83d580ae |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/1/artifact/out/diff-checkstyle-root.txt |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/1/artifact/out/whitespace-eol.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/1/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/1/testReport/ |
| Max. process+thread count | 317 (vs. ulimit of 10000) |
| modules | C: hbase-annotations hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-it . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 18 new or modified test files.  |
||| _ HBASE-22514 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 42s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 11s |  HBASE-22514 passed  |
| +1 :green_heart: |  compile  |   3m 17s |  HBASE-22514 passed  |
| +1 :green_heart: |  checkstyle  |   3m  0s |  HBASE-22514 passed  |
| +1 :green_heart: |  shadedjars  |   5m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 48s |  HBASE-22514 passed  |
| +0 :ok: |  spotbugs  |   1m 25s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 13s |  branch/hbase-annotations no findbugs output file (findbugsXml.xml)  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 16s |  the patch passed  |
| +1 :green_heart: |  cc  |   3m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 17s |  the patch passed  |
| -1 :x: |  checkstyle  |   3m  1s |  root: The patch generated 1 new + 60 unchanged - 1 fixed = 61 total (was 61)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   5m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 27s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   8m 39s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 42s |  the patch passed  |
| +0 :ok: |  findbugs  |   0m 11s |  hbase-annotations has no data from findbugs  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 253m 16s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   2m 47s |  The patch does not generate ASF License warnings.  |
|  |   | 391m 23s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.security.access.TestSnapshotScannerHDFSAclController |
|   | hadoop.hbase.rsgroup.TestMigrateRSGroupInfo |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/813 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux 02a7234fc425 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-813/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / eb83d580ae |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/2/artifact/out/diff-checkstyle-root.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/2/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/2/testReport/ |
| Max. process+thread count | 4369 (vs. ulimit of 10000) |
| modules | C: hbase-annotations hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-it . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 18 new or modified test files.  |
||| _ HBASE-22514 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 44s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 13s |  HBASE-22514 passed  |
| +1 :green_heart: |  compile  |   2m 56s |  HBASE-22514 passed  |
| +1 :green_heart: |  checkstyle  |   2m 36s |  HBASE-22514 passed  |
| +1 :green_heart: |  shadedjars  |   4m 40s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 50s |  HBASE-22514 passed  |
| +0 :ok: |  spotbugs  |   1m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 16s |  branch/hbase-annotations no findbugs output file (findbugsXml.xml)  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 58s |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 58s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 58s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 31s |  root: The patch generated 0 new + 60 unchanged - 1 fixed = 60 total (was 61)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   4m 41s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 43s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   8m 17s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 54s |  the patch passed  |
| +0 :ok: |  findbugs  |   0m 12s |  hbase-annotations has no data from findbugs  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 178m 54s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   3m 48s |  The patch does not generate ASF License warnings.  |
|  |   | 304m 50s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.rsgroup.TestMigrateRSGroupInfo |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/813 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux c7e4f9d33406 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-813/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 1eb2ded8b3 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/3/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/3/testReport/ |
| Max. process+thread count | 4918 (vs. ulimit of 10000) |
| modules | C: hbase-annotations hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-it . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java [hunk 1]
```diff
@@ -2315,15 +2315,15 @@ boolean snapshotCleanupSwitch(final boolean on, final boolean synchronous)
    * @param servers set of servers to remove
    * @throws IOException if a remote or network exception occurs
    */
-  void removeRSGroup(Set<Address> servers) throws IOException;
+  void removeServers(Set<Address> servers) throws IOException;
```
Comment (infraio) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java]: removeServersFromRSGroup? And no need rsgroup as the parameter?

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java [hunk 2]
```diff
@@ -2315,15 +2315,15 @@ boolean snapshotCleanupSwitch(final boolean on, final boolean synchronous)
...
 
   /**
    * Move given set of servers to the specified target RegionServer group
    * @param servers set of servers to move
    * @param targetGroup the group to move servers to
    * @throws IOException if a remote or network exception occurs
    */
-  void moveToRSGroup(Set<Address> servers, String targetGroup) throws IOException;
+  void moveServers(Set<Address> servers, String targetGroup) throws IOException;
```
Comment (infraio) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java]: moveServersToRSGroup?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java [hunk 3]
```diff
@@ -2860,6 +2866,94 @@ private boolean shouldSubmitSCP(ServerName serverName) {
...
+      if (master.getMasterCoprocessorHost() != null) {
+        master.getMasterCoprocessorHost().preGetRSGroupInfoOfServer(hp);
+      }
+      RSGroupInfo rsGroupInfo = master.getRSGroupInfoManager().getRSGroupOfServer(hp);
+      GetRSGroupInfoOfServerResponse resp;
+      if (rsGroupInfo != null) {
+        resp = GetRSGroupInfoOfServerResponse.newBuilder()
+          .setRSGroupInfo(ProtobufUtil.toProtoGroupInfo(rsGroupInfo)).build();
+      } else {
+        resp = GetRSGroupInfoOfServerResponse.getDefaultInstance();
```
Comment (infraio) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java]: When return this, it means the table not exist or the server not exist?

File: hbase-it/src/test/java/org/apache/hadoop/hbase/rsgroup/IntegrationTestRSGroup.java [hunk 4]
```diff
@@ -41,24 +41,23 @@
...
-      //set shared configs
-      admin = TEST_UTIL.getAdmin();
-      cluster = TEST_UTIL.getHBaseClusterInterface();
-      rsGroupAdmin = new VerifyingRSGroupAdminClient(new RSGroupAdminClient(TEST_UTIL.getConnection()),
-          TEST_UTIL.getConfiguration());
+      ((IntegrationTestingUtility) TEST_UTIL).initializeCluster(NUM_SLAVES_BASE);
+      // set shared configs
+      ADMIN = TEST_UTIL.getAdmin();
+      CLUSTER = TEST_UTIL.getHBaseClusterInterface();
+      RS_GROUP_ADMIN_CLIENT = new RSGroupAdminClient(TEST_UTIL.getConnection());
```
Comment (infraio) [inline hbase-it/src/test/java/org/apache/hadoop/hbase/rsgroup/IntegrationTestRSGroup.java]: Still need RS_GROUP_ADMIN_CLIENT?

Comment (Apache9) [reply hbase-it/src/test/java/org/apache/hadoop/hbase/rsgroup/IntegrationTestRSGroup.java]: Can be a follow-on issue. Now when calling getRSGroup in UT, we assume that the RSGroupInfo contains all the tables in this group. Plan to add a method for getting the tables within a group and rewrite these UTs. See HBASE-23276.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java (hunk 3 above)
Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java]: I think here we do not care whether the server is online or not? Just means there is no group which contains this server.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java (hunk 1 above)
Comment (Apache9) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java]: A server can only belong to one group, the the group here is redundant? In the implementation, we will check whether these servers are from the same group.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 18 new or modified test files.  |
||| _ HBASE-22514 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 43s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 35s |  HBASE-22514 passed  |
| +1 :green_heart: |  compile  |   3m 14s |  HBASE-22514 passed  |
| +1 :green_heart: |  checkstyle  |   3m  0s |  HBASE-22514 passed  |
| +1 :green_heart: |  shadedjars  |   5m  2s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 41s |  HBASE-22514 passed  |
| +0 :ok: |  spotbugs  |   1m 34s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 13s |  branch/hbase-annotations no findbugs output file (findbugsXml.xml)  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 14s |  the patch passed  |
| +1 :green_heart: |  cc  |   3m 14s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 14s |  the patch passed  |
| -1 :x: |  checkstyle  |   2m 57s |  root: The patch generated 2 new + 60 unchanged - 1 fixed = 62 total (was 61)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   5m  1s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 15s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   8m 46s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 43s |  the patch passed  |
| +0 :ok: |  findbugs  |   0m 11s |  hbase-annotations has no data from findbugs  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 228m 52s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   3m 15s |  The patch does not generate ASF License warnings.  |
|  |   | 365m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/813 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux 77f644fdb900 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-813/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 1eb2ded8b3 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/4/artifact/out/diff-checkstyle-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/4/testReport/ |
| Max. process+thread count | 4346 (vs. ulimit of 10000) |
| modules | C: hbase-annotations hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-it . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 28s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 18 new or modified test files.  |
||| _ HBASE-22514 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 41s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 40s |  HBASE-22514 passed  |
| +1 :green_heart: |  compile  |   3m 14s |  HBASE-22514 passed  |
| +1 :green_heart: |  checkstyle  |   2m 59s |  HBASE-22514 passed  |
| +1 :green_heart: |  shadedjars  |   5m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 40s |  HBASE-22514 passed  |
| +0 :ok: |  spotbugs  |   1m 25s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 13s |  branch/hbase-annotations no findbugs output file (findbugsXml.xml)  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 21s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 12s |  the patch passed  |
| +1 :green_heart: |  cc  |   3m 12s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 12s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 53s |  root: The patch generated 0 new + 60 unchanged - 1 fixed = 60 total (was 61)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   5m  4s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 10s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   8m 42s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 43s |  the patch passed  |
| +0 :ok: |  findbugs  |   0m 12s |  hbase-annotations has no data from findbugs  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 258m 45s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   3m 20s |  The patch does not generate ASF License warnings.  |
|  |   | 394m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/813 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool xml |
| uname | Linux 410eb2e68de6 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-813/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 1eb2ded8b3 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/6/testReport/ |
| Max. process+thread count | 4910 (vs. ulimit of 10000) |
| modules | C: hbase-annotations hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-it . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-813/6/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**